envoy/http/alternate_protocols_cache.h

```diff
@@ -87,6 +87,22 @@ class AlternateProtocolsCache {
     MonotonicTime expiration_;
   };
 
+  class Http3StatusTracker {
+  public:
+    virtual ~Http3StatusTracker() = default;
+
+    // Returns true if HTTP/3 is broken.
+    virtual bool isHttp3Broken() const PURE;
+    // Returns true if HTTP/3 is confirmed to be working.
+    virtual bool isHttp3Confirmed() const PURE;
+    // Marks HTTP/3 broken for a period of time, subject to backoff.
+    virtual void markHttp3Broken() PURE;
+    // Marks HTTP/3 as confirmed to be working and resets the backoff timeout.
+    virtual void markHttp3Confirmed() PURE;
+    // Returns string form of the HTTP/3 status.
+    virtual std::string statusToStringForCache() PURE;
```

Comment:
FWIW, in Chrome the HTTP/3 status is not persisted. We're welcome to do something different in Envoy, of course, but we might not need to.

Reply:
Hmm, good point! Is the life time of alternate protocols cache the same as Envoy? Does the difference of whether to persist or not only matter at the first request to each host?

Reply:
I see. I removed the To/FromString transformation for now.